pennylane/measurements/state.py

```diff
@@ -211,4 +212,7 @@ def process_state(self, state: Sequence[complex], wire_order: Wires):
         # pylint:disable=redefined-outer-name
         wire_map = dict(zip(wire_order, range(len(wire_order))))
         mapped_wires = [wire_map[w] for w in self.wires]
-        return qml.math.reduce_statevector(state, indices=mapped_wires)
+        kwargs = {"indices": mapped_wires, "c_dtype": "complex128"}
+        if not qml.math.is_abstract(state) and qml.math.any(qml.math.iscomplex(state)):
+            kwargs["c_dtype"] = state.dtype
```

Comment:
why not always updating the `c_dtype` independent from the requested meas?
```suggestion
        kwargs["c_dtype"] = state.dtype
```

Reply:
I think the goal is to return the complex type of the matching precision, but some states are integer, float, etc.

Reply:
gotta! :+1: